File: Examples/test-suite/ocaml/default_arg_values_runme.ml
open Swig
open Default_arg_values

let _ =
  let d = new_Display '() in
  assert (d -> draw1 () as float = 0.);
  let arg = C_float 12. in
  assert (d -> draw1 (arg) as float = 12.);
  let arg = C_float 123. in
  let p = _createPtr '(arg) in
  assert (d -> draw2 () as float = 0.);
  assert (d -> draw2 (p) as float = 123.);
  assert (d -> bool0 () as bool = false);
  assert (d -> bool1 () as bool = true);
  assert (d -> mybool0 () as bool = false);
  assert (d -> mybool1 () as bool = true)
;;

File: Examples/test-suite/javascript/default_arg_values_runme.js
var default_arg_values = require("default_arg_values");

var d = new default_arg_values.Display();

if (d.draw1() != 0) {
    throw new Error;
}

if (d.draw1(12) != 12) {
    throw new Error;
}

p = default_arg_values.createPtr(123);
if (d.draw2() != 0) {
    throw new Error;
}

if (d.draw2(p) != 123) {
    throw new Error;
}

if (d.bool0() != false || typeof d.bool0() != typeof false) {
    throw new Error;
}

if (d.bool1() != true || typeof d.bool1() != typeof true) {
    throw new Error;
}

if (d.mybool0() != false || typeof d.mybool0() != typeof false) {
    throw new Error;
}

if (d.mybool1() != true || typeof d.mybool1() != typeof false) {
    throw new Error;
}
